src/azure-cli/azure/cli/command_modules/keyvault/_params.py

```diff
@@ -399,7 +647,46 @@ class CLIJsonWebKeyOperation(str, Enum):
         c.argument('admin_email')
         c.argument('admin_phone')
 
-    for scope in ['list', 'list-deleted', 'list-versions']:
-        with self.argument_context('keyvault certificate {}'.format(scope)) as c:
+    for item in ['list', 'list-deleted', 'list-versions']:
+        with self.argument_context('keyvault certificate {}'.format(item)) as c:
             c.argument('maxresults', options_list=['--maxresults'], type=int)
+
+    with self.argument_context('keyvault role') as c:
+        c.argument('scope',
+                   help='scope at which the role assignment or definition applies to, '
+                        'e.g., "/" or "/keys" or "/keys/{keyname}"')
+
+    with self.argument_context('keyvault role', arg_group='Id') as c:
+        c.argument('hsm_name', hsm_url_type)
+        c.argument('identifier', options_list=['--id'],
+                   help='Id of the HSM. If specified all other \'Id\' arguments should be omitted.',
+                   validator=process_hsm_name)
+
+    with self.argument_context('keyvault role assignment') as c:
+        c.argument('role_assignment_name', options_list=['--name', '-n'], help='Name of the role assignment.')
+        c.argument('assignee', help='represent a user, group, or service principal. '
+                                    'supported format: object id, user sign-in name, or service principal name')
+        c.argument('assignee_object_id',
+                   help='Use this parameter instead of \'--assignee\' to bypass graph permission issues. '
+                        'This parameter only works with object ids for users, groups, service principals, and '
+                        'managed identities. For managed identities use the principal id. For service principals, '
+                        'use the object id and not the app id.')
+        c.argument('ids', nargs='+', help='space-separated role assignment ids')
+        c.argument('role', help='role name or id')
```

Comment:
Just curious, may I ask why Key Vault has the separate RBAC command?
Why not use `az role xxx`?

Reply:
@zhoxing-ms Good question. I'm not very clear about the exact motivation of the service team. One possible reason is that they didn't want to expose HSM related access controls to the outer world (AAD), so they implemented some similar kinds of stuff which can under their own control.